tutorials/shaders/shader_reference/shader_preprocessor.rst

```diff
@@ -0,0 +1,141 @@
+.. _doc_shader_preprocessor:
+
+Shader Preprocessor
+===================
+
+Shader preprocessor belongs as optional step before shader compilation of text shaders in Godot.
+If you don't need it, you may not use it, but it contains some useful macros which may speed up your productivity.
+
+#define
+^^^^^^^
+\ **Syntax:** `#define identifier <replacement_code>`.
+
+Defines the identifier after that directive as macro, and replaces all successive occurrence of it to the replacement code in the shader.
+If the replacement code is not defined, it may only been used within `#ifdef` or `#ifndef` directives.
+
+.. code-block:: glsl
+
+    shader_type spatial;
+
+    #define USE_MY_COLOR
+    #define MY_COLOR vec3(1, 0, 0)
+
+    void fragment() {
+    #ifdef USE_MY_COLOR
+        ALBEDO = MY_COLOR;
+    #endif
+    }
+
+#if
+^^^
+\ **Syntax:** `#if condition`.
+
+The `#if` directive checks the condition and if it evaluates to a non-zero value, the controlled code block is included and skipped otherwise.
```

Comment:
```suggestion
The `#if` directive checks the condition and if it evaluates to a non-zero value, the code block is included, otherwise it is skipped.
```